### PR TITLE
Update IndexSettingProvider#getAdditionalIndexSettings() signature

### PR DESCRIPTION
**Pull request by martijnvg**
With logsdb another index mode is available, the isTimeSeries parameter is limiting. Instead, we should just push down the index mode from template to index settings provider.

Follow up from #113451
Relates to #113583

### CODE REVIEW

**Comment by martijnvg** [comment]
Note to reviewer: a downstream index settings provider also needs to be updated to use the new signature.

**Review by lkts — APPROVED** [review]

**Comment by lkts** [comment]
Serverless failure seems legit.

File: modules/data-streams/src/main/java/org/elasticsearch/datastreams/DataStreamIndexSettingsProvider.java [hunk 1]
```diff
@@ -70,6 +70,7 @@ public Settings getAdditionalIndexSettings(
             // First backing index is created and then data stream is rolled over (in a single cluster state update).
             // So at this point we can't check index_mode==time_series,
             // so checking that index_mode==null|standard and templateIndexMode == TIME_SERIES
+            boolean isTimeSeries = templateIndexMode == IndexMode.TIME_SERIES;
             boolean migrating = dataStream != null
```
Comment (kkrik-es) [inline modules/data-streams/src/main/java/org/elasticsearch/datastreams/DataStreamIndexSettingsProvider.java:74]: Nit: `migratingTimeSeries`?

File: modules/data-streams/src/main/java/org/elasticsearch/datastreams/DataStreamIndexSettingsProvider.java [hunk 2]
```diff
@@ -70,6 +70,7 @@ public Settings getAdditionalIndexSettings(
             // First backing index is created and then data stream is rolled over (in a single cluster state update).
             // So at this point we can't check index_mode==time_series,
             // so checking that index_mode==null|standard and templateIndexMode == TIME_SERIES
+            boolean isTimeSeries = templateIndexMode == IndexMode.TIME_SERIES;
```
Comment (kkrik-es) [inline modules/data-streams/src/main/java/org/elasticsearch/datastreams/DataStreamIndexSettingsProvider.java:73]: Check for null?

File: server/src/test/java/org/elasticsearch/cluster/metadata/MetadataTests.java [hunk 3]
```diff
@@ -2435,7 +2436,7 @@ public void testIsTimeSeriesTemplate() throws IOException {
                 .dataStreamTemplate(new ComposableIndexTemplate.DataStreamTemplate())
                 .build();
             Metadata m = Metadata.builder().put("component_template_1", componentTemplate).put("index_template_1", indexTemplate).build();
-            assertThat(m.isTimeSeriesTemplate(indexTemplate), is(true));
+            assertThat(m.retrieveIndexModeFromTemplate(indexTemplate), is(IndexMode.TIME_SERIES));
```
Comment (kkrik-es) [inline server/src/test/java/org/elasticsearch/cluster/metadata/MetadataTests.java:2439]: Shall we add tests where it returns another mode and null?

**Review by kkrik-es — APPROVED** [review]

File: modules/data-streams/src/main/java/org/elasticsearch/datastreams/DataStreamIndexSettingsProvider.java (hunk 2 above)
Comment (martijnvg) [reply modules/data-streams/src/main/java/org/elasticsearch/datastreams/DataStreamIndexSettingsProvider.java:73]: if `templateIndexMode` is null then this evaluates false, which is expected. So I think this is good?

**Review by salvatore-campagna — APPROVED** [review]

**Pull request merged**